src/LocalPath.cpp

```diff
@@ -104,7 +104,10 @@ LocalPath(Path file) noexcept
 {
   assert(file != nullptr);
 
-  return AllocatedPath::Build(GetPrimaryDataPath(), file);
+  if (file.IsAbsolute())
+    return AllocatedPath(file);
```

Comment:
This defeats the point of calling `LocalPath()`. Your commit message doesn't explain why this is necessary.

Reply:
Ok, I see... 
In former time (~6 month ago...) I had a lot of trouble with the '-datapath' parameter in the commandline. There was no way to set it to the root directory, f.e. if I want to use a data dir in an other partition. Now it seems to work, but only if I set the profile file with -profile in the same directory. Without the profile setting XCSoar is loading all stuff from the default directory ;-(

Here I want to prevent the  file upload with an absolute path - but this maked trouble because I did a wrong change in src/Cloud/weglide/UploadFlight.cpp, line 47 like you mentioned in the next post! 
So I will review this completely up to the next push!

But the (new) problems with datapath and profile in the commandline are still there yet...

Reply:
I removed this commit - and I will really check the issue with the datapath again. For this PR it isn't needed!